src/main/java/aztech/modern_industrialization/api/pipe/item/SpeedUpgrade.java

```diff
@@ -24,22 +24,31 @@
 package aztech.modern_industrialization.api.pipe.item;
 
 import aztech.modern_industrialization.MIItem;
-import java.util.IdentityHashMap;
+import aztech.modern_industrialization.compat.kubejs.KubeJSProxy;
+import java.util.HashMap;
 import java.util.Map;
+import net.minecraft.core.registries.BuiltInRegistries;
+import net.minecraft.resources.ResourceLocation;
 import net.minecraft.world.item.Item;
 
 /**
  * A speed upgrade for an item pipe
  */
 public class SpeedUpgrade {
```

Comment:
As a general remark once the others are addressed: this needs documentation in the `docs/` folder.